source/postprocess/gravity_point_values.cc

```diff
@@ -0,0 +1,322 @@
+                  density_JxW[local_cell_number * number_quadrature_points_cell + q] = out.densities[q] * fe_values.JxW(q);
+                  position_point[local_cell_number * number_quadrature_points_cell + q] = position_point_cell[q];
+                }
+              ++local_cell_number;
+            }
+        }
+
+      // loop on r - satellite position [r, ,  ]
+      for (unsigned int h=0; h < number_points_radius; ++h)
+        {
+          std_cxx11::array<double,dim> satellite_coordinate;
+          satellite_coordinate[0] = minimum_radius + ((maximum_radius - minimum_radius) / number_points_radius) * h;
+
+          // loop on phi - satellite position [ , phi ,]
+          for (unsigned int i=0; i < number_points_longitude; ++i)
+            {
+              satellite_coordinate[1] = (minimum_longitude + ((maximum_longitude - minimum_longitude) / number_points_longitude) * i) * numbers::PI / 180.;
+
+              // loop on theta - satllite position [ , , theta]
+              for (unsigned int j=0; j < number_points_latitude; ++j)
+                {
+                  satellite_coordinate[2] = (minimum_latitude + ((maximum_latitude - minimum_latitude) / number_points_latitude) * j) * numbers::PI / 180.;
+                  Tensor<1,dim> local_g;
+                  double local_U = 0;
+
+                  // The spherical coordinates are shifted into cartesian to allow simplification in the mathematical equation.
+                  const Point<dim> position_satellite = Utilities::Coordinates::spherical_to_cartesian_coordinates<dim>(satellite_coordinate);
+
+                  // For each point (i.e. satellite), the fourth integral goes over cells and quadrature
+                  // points to get the unique distance between those, indispensable to calculate
+                  // gravity vector components x,y,z, and potential.
+                  cell = this->get_dof_handler().begin_active();
+                  local_cell_number = 0;
+                  for (; cell!=endc; ++cell)
+                    {
+                      if (cell->is_locally_owned())
+                        {
+                          for (unsigned int q = 0; q < number_quadrature_points_cell; ++q)
+                            {
+                              double dist = (position_satellite - position_point[local_cell_number * number_quadrature_points_cell + q]).norm();
+                              double KK = G * density_JxW[local_cell_number * number_quadrature_points_cell + q] / pow(dist,3);
+                              local_g += KK * (position_satellite - position_point[local_cell_number * number_quadrature_points_cell + q]);
+                              local_U -= G * density_JxW[local_cell_number * number_quadrature_points_cell + q] / dist;
+                            }
+                          ++local_cell_number;
+                        }
+                    }
+
+                  // assemble gravity component results from the mpi
```

Comment:
we use `assemble` only when assembling matrices. In this case I would describe it as `Sum gravity components over global domain`